src/DynamoCoreWpf/Utilities/JobDebouncer.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+namespace Dynamo.Wpf.Utilities
+{
+    public static class JobDebouncer
```

Comment:
- can you add some more description to this utility, as to when this should be used? I understand we all know that answer already, but would like to document so everyone and new developers on team also learn to use it. 

- There are also a few other instances using deboucer in our code, can you unify the usage?

Reply:
Added a class summary description. The PR as currently is already unifies both library (sidebar) and incontext node search into using the debouncer, but I can't find more instances of a debouncer being used in c# code (there are some in html/js webview code though).